orttraining/orttraining/training_api/include/onnxruntime_training_c_api.h

```diff
@@ -0,0 +1,41 @@
+// This file contains c apis for on device training
+// This file should never be included standalone
+// It is included from within core/session/onnxruntime_c_api.h when
+// on device training is enabled
+// These apis can be moved to core/session/onnxruntime_c_api.h once they stabilize
+
+// DO NOT UNCOMMENT
+//#include "core/session/onnxruntime_c_api.h"
+
+ORT_API2_STATUS(LoadCheckpoint, _In_ const ORTCHAR_T* checkpoint_path, _Outptr_ OrtCheckpointState** checkpoint_state);
```

Comment:
will add comments for each api right before merging (similar to what we have in the master c_api.h)